src/Kernel/FluentValidationExtensions/ValidateAndThrowExtension.cs

```diff
@@ -0,0 +1,30 @@
+﻿using FluentValidation;
+using System.Linq;
+
+namespace LT.DigitalOffice.Kernel.FluentValidationExtensions
+{
+    /// <summary>
+    /// Provides ValidateAndThrow method for extend FluentValidation package.
+    /// </summary>
+    public static class ValidateAndThrowExtension
+    {
+        /// <summary>
+		/// Performs validation and then throws an exception if validation fails.
+		/// </summary>
+		/// <param name="validator">The validator this method is extending.</param>
+		/// <param name="instance">The instance of the type we are validating.</param>
+		/// <param name="ruleSet">Optional: a ruleset when need to validate against.</param>
+        public static void ValidateAndThrowCustom<T>(this IValidator<T> validator, T instance, string ruleSet = null)
+        {
+            var result = validator.Validate(instance, options => options.IncludeRuleSets(ruleSet));
+
+            if (result != null && !result.IsValid)
+            {
+                var messages = result.Errors.Select(x => x.ErrorMessage);
+                string message = string.Join("\n", messages);
```

Comment:
Why not StringBuilder?

Reply:
Join is approximately as fast as StringBuilder

Reply:
https://stackoverflow.com/a/585907

Reply:
but it clogs memory because new objects are created in memory, this is bad

Reply:
Are you sure about new copies? As I understood it, this method doesn't do it

Reply:
https://stackoverflow.com/a/585897

Reply:
This is complicated quiestion, there is no simple answer. Context is really important, right here we already have IEnumerable<string> filled with strings, so string.Join will be much more efficient and faster.
Plz check this article
https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=netcore-3.1#the-string-and-stringbuilder-types